plugin/evm/vm.go

```diff
@@ -1218,15 +1226,41 @@ func (vm *VM) setAppRequestHandlers() {
 			},
 		},
 	)
+	if err := vm.RegisterLeafRequestHandler(message.StateTrieNode, "sync_state_trie_leaves", evmTrieDB, message.StateTrieKeyLength, true); err != nil {
+		return err
+	}
+	// Register atomic trieDB for serving atomic leafs requests.
+	if err := vm.RegisterLeafRequestHandler(atomic.AtomicTrieNode, "sync_atomic_trie_leaves", vm.atomicTrie.TrieDB(), atomic.AtomicTrieKeyLength, false); err != nil {
+		return err
+	}
+
 	networkHandler := newNetworkHandler(
 		vm.blockChain,
 		vm.chaindb,
-		evmTrieDB,
-		vm.atomicTrie.TrieDB(),
 		vm.warpBackend,
 		vm.networkCodec,
+		vm.leafRequestTypeConfigs,
 	)
 	vm.Network.SetRequestHandler(networkHandler)
+	return nil
+}
+
+func (vm *VM) RegisterLeafRequestHandler(nodeType message.NodeType, metricName string, trieDB *triedb.Database, trieKeyLen int, useSnapshot bool) error {
```

Comment:
Hmm I wonder if we should register an interface rather than nodeType directly. I feel like there is definitely room to improve this PR in general, but did not want to break it anything further.